### ISSUE TITLE
Allow Parallel Garbage Collection of Blocks

### ISSUE THREAD

**Issue by kderme** [input-output-hk/ouroboros-network#1716]
After recent changes, which disallow garbage collecting the most recent file in the volatile db, we can allow a thread to garbage collect the db, while another thread inserts blocks, in parallel.

**Comment by mrBliss** [comment]
Related: #1866 and #1919. For simplicity, we won't do GCs in parallel with appending. Also, as of #1923 we're doing much fewer GCs, so it wouldn't make that big a difference anymore.
